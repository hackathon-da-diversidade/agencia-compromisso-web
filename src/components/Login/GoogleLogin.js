import React from 'react';
import Login from './Login';

const getGoogleAuthUri = () =>
  fetch('/auth')
    .then(res => res.text())
    .then(uri => (window.location.href = uri));

export default () => {
  return <Login loginAction={getGoogleAuthUri} />;
};
