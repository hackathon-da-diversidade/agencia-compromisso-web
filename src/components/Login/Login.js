import React, { Component } from 'react';
import icon from '../../assets/icon.png';
import banner from '../../assets/banner.png';
import { withRouter } from 'react-router-dom';

const SignIn = React.lazy(() => import('./SignIn'));

class Login extends Component {
  render() {
    return (
      <>
        <div className="menu">
          <img
            className="icon"
            src={icon}
            alt="ícone estilizado com o perfil de uma mulher com uma fita métrica no black power"
          />
          <img
            className="banner"
            src={banner}
            alt="banner da Agência Compromisso"
          />
          <ul className="agencia-btn-group">
            <li>
              <SignIn history={this.props.history} />
            </li>
          </ul>
        </div>
      </>
    );
  }
}

export default withRouter(Login);
