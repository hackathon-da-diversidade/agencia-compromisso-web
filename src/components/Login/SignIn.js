import { useAuth } from 'reactfire';
import React from 'react';
import * as firebase from 'firebase';
import AuthorizeAPI from '../../api/authorizeAPI';
import googleIcon from '../../assets/googleIconSmall.png';

const SignIn = ({ history }) => {
  const auth = useAuth();

  const signIn = async () => {
    const userCredential = await auth.signInWithPopup(
      new firebase.auth.GoogleAuthProvider(),
    );

    const status = await AuthorizeAPI.get(userCredential.user.email);

    if (status === 200) {
      return history.push('/menu');
    }

    await auth.signOut();
  };

  return (
    <button onClick={signIn} className="btn btn-secondary btn-raised">
      <img className="google-icon" src={googleIcon} alt="Ã­cone do Google"/>
      Login
    </button>
  );
};

export default SignIn;
