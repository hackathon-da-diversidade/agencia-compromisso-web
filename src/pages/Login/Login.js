import React from 'react';
import { withRouter } from 'react-router-dom';
import { useAuth } from 'reactfire';
import * as firebase from 'firebase/app';
import 'firebase/auth';
import Logo from 'components/Logo/Logo';
import Menu from 'components/Menu/Menu';
import AuthorizeAPI from 'api/authorizeAPI';
import googleIcon from 'assets/googleIconSmall.png';

const Login = ({ history }) => {
  const auth = useAuth();

  const signIn = async () => {
    const userCredential = await auth.signInWithPopup(
      new firebase.auth.GoogleAuthProvider()
    );

    const status = await AuthorizeAPI.get(userCredential.user.email);

    if (status === 200) {
      return history.push('/');
    }

    await auth.signOut();
  };

  return (
    <>
      <Logo />
      <Menu
        menuOptions={[
          {
            text: 'Login',
            onClick: signIn,
            icon: <img src={googleIcon} alt="Ã­cone do Google"></img>,
          },
        ]}
      />
    </>
  );
};

export default withRouter(Login);
