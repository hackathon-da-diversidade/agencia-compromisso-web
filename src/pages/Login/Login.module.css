.GoogleIcon {
    float: left;
    padding-top: 2px;
    padding-left: 5px;
}