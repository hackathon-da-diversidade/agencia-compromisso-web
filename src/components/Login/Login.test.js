import React from 'react';
import { render, fireEvent, cleanup } from '@testing-library/react';
import '@testing-library/jest-dom/extend-expect';

import Login from './Login';

afterEach(cleanup);

it('should render the login button', () => {
  const { getByRole } = render(<Login loginAction={() => {}} />);
  expect(getByRole('button')).toHaveTextContent('Login');
});

it('Should click the login on the button', () => {
  const mockAction = jest.fn();
  const { getByText } = render(<Login loginAction={mockAction} />);
  fireEvent.click(getByText('Login'));

  expect(mockAction.mock.calls.length).toEqual(1);
});

it('matches the snapshot', () => {
  const { asFragment } = render(<Login loginAction={() => {}} />);
  expect(asFragment()).toMatchSnapshot();
});
