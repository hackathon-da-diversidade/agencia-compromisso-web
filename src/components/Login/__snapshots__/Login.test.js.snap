// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches the snapshot 1`] = `
<DocumentFragment>
  <div
    class="menu"
  >
    <img
      alt="ícone estilizado com o perfil de uma mulher com uma fita métrica no black power"
      class="icon"
      src="icon.png"
    />
    <img
      alt="banner da Agência Compromisso"
      class="banner"
      src="banner.png"
    />
    <ul
      class="agencia-btn-group"
    >
      <li>
        <button
          class="btn btn-secondary btn-raised"
        >
          <img
            alt="ícone do Google"
            class="google-icon"
            src="googleIconSmall.png"
          />
          Login
        </button>
      </li>
    </ul>
  </div>
</DocumentFragment>
`;
